drivers/abp2/abp2.c

```diff
@@ -0,0 +1,242 @@
+        loopsTimeout--;
+        ztimer_sleep(ZTIMER_MSEC, 1);   /* wait 1ms */
+    }
+    if (!loopsTimeout) {
+        return -ETIMEDOUT;
+    }
+    /* check sensor errors: */
+    if (status & ABP2_STATUS_MASK) {
+        return -ENODATA;
+    }
+    /* read ADC conversion results: */
+    res = abp2_read_raw(dev, &rawData);
+    if (res) {
+        return res;
+    }
+    /* convert to physical quantities: */
+    *press = abp2_pressure(dev, &rawData);
+    *temp = abp2_temperature(dev, &rawData);
+
+    return 0;
+}
+
+int abp2_read_nb(const abp2_t *dev, int32_t *press, int32_t *temp)
+{
+    int res = 0;
+    abp2_raw_t rawData;
+
+    /* start an ADC conversion and read previous results: */
+    res = abp2_measure_read(dev, &rawData);
+    if (res) {
+        return res;
+    }
+    /* convert to physical quantities: */
+    *press = abp2_pressure(dev, &rawData);
+    *temp = abp2_temperature(dev, &rawData);
+
+    return 0;
+}
+
+uint8_t abp2_getstatus(const abp2_t *dev)
+{
+    uint8_t status = 0xFF;      /* sensor status byte */
+
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    status = spi_transfer_byte(dev->params->spi, dev->params->cs, false, ABP2_CMD_NOP);
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+
+    return status;
+}
+
+int abp2_read_raw(const abp2_t *dev, abp2_raw_t *raw_values)
+{
+    uint8_t status = 0xFF;      /* sensor status byte */
+    uint8_t data_rx[ABP2_RX_BUF_LEN];
+
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs, false, data_tx_nop_read, data_rx,
+                       sizeof(data_tx_nop_read));
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+    status = data_rx[0];
+    if (status & ABP2_STATUS_MASK) {
+        return -ENODATA;
+    }
+    raw_values->cntPress = data_rx[1] << 16 | data_rx[2] << 8 | data_rx[3];
+    raw_values->cntTemp = data_rx[4] << 16 | data_rx[5] << 8 | data_rx[6];
+
+    return 0;
+}
+
+int abp2_measure_read(const abp2_t *dev, abp2_raw_t *raw_values)
+{
+    uint8_t status = 0xFF;     /* sensor status byte */
+    int res = 0;
+    uint8_t data_rx[ABP2_RX_BUF_LEN];
+
+#if defined(MODULE_ABP2_SPI)
+    spi_acquire(dev->params->spi, dev->params->cs, SPI_MODE_0, dev->params->clk);
+    spi_transfer_bytes(dev->params->spi, dev->params->cs, false, data_tx_meas_read, data_rx,
+                       sizeof(data_tx_meas_read));
+    spi_release(dev->params->spi);
+#else
+#pragma message("implement I2C code here")
+#endif
+    status = data_rx[0];
+    if (status & ABP2_STATUS_MASK) {
+        return -ENODATA;
+    }
+    raw_values->cntPress = data_rx[1] << 16 | data_rx[2] << 8 | data_rx[3];
+    raw_values->cntTemp = data_rx[4] << 16 | data_rx[5] << 8 | data_rx[6];
```

Comment:
```suggestion
    raw_values->cntPress = (uint32_t)data_rx[1] << 16 | data_rx[2] << 8 | data_rx[3];
    raw_values->cntTemp = (uint32_t)data_rx[4] << 16 | data_rx[5] << 8 | data_rx[6];
```

Reply:
Nice little bug... I'll cast every `data_rx` that is left-shifted on these 2 lines.

Reply:
murdock (our CI) found it https://ci.riot-os.org/results/985f6c7a96084e44936b775aa4737887/output/builds/tests/drivers/abp2/atmega256rfr2-xpro:gnu.txt

see

https://ci.riot-os.org/results/985f6c7a96084e44936b775aa4737887/output/builds/tests/drivers/abp2/atmega256rfr2-xpro:gnu.txt